corehq/apps/users/user_data.py

```diff
@@ -188,3 +190,23 @@ class SQLUserData(models.Model):
     class Meta:
         unique_together = ("user_id", "domain")
         indexes = [models.Index(fields=['user_id', 'domain'])]
+
+
+def prime_user_data_caches(users, domain):
+    """Lookup and prime caches of user data in chunks"""
```

Comment:
The name of this function implies that caches are primed when it is called, but nothing will happen unless the generator is iterated. What do you think of adding note about that in the docstring?

Reply:
Good point, will do

Reply:
I wrote it initially as something that would evaluate all up front, but lazy eval seemed more appropriate for an "enrich" type utility.  Let me know if you have a better idea for the name.